examples/cluster-with-hpa.yaml

```diff
@@ -0,0 +1,45 @@
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+apiVersion: spark.apache.org/v1alpha1
+kind: SparkCluster
+metadata:
+  name: cluster-with-hpa
+spec:
+  runtimeVersions:
+    sparkVersion: "4.0.0-preview2"
+  clusterTolerations:
+    instanceConfig:
+      initWorkers: 3
+      minWorkers: 1
+      maxWorkers: 3
+  workerSpec:
+    workerStatefulSetSpec:
+      template:
+        spec:
+          containers:
+          - name: worker
+            resources:
+              requests:
+                cpu: "3"
+                memory: "3Gi"
+              limits:
+                cpu: "3"
+                memory: "3Gi"
```

Comment:
Hmm, this definition is enough for hpa? Don't we need to specify scaling behavior?

Reply:
To @viirya , this is not HPA part. This is a minimum pod spec which HPA requires later.

Previously, the user creates HPA manually, in this PR, HPA part is hard-coded here for now. It will be revised later by allowing template.
- https://github.com/apache/spark-kubernetes-operator/pull/135/files#diff-2366f4d987646a8a1cd7d555ab69d582f647bfd86c3fdc773506f6e2743b3da2R313-R326

Reply:
Okay. Then I expect that we will add more into this example after revising later.